projects/storefrontlib/src/cms-components/myaccount/my-interests/my-interests.component.ts

```diff
@@ -0,0 +1,105 @@
+import { Component, OnInit, OnDestroy } from '@angular/core';
+import { Observable, combineLatest } from 'rxjs';
+import {
+  UserInterestsService,
+  ProductInterestSearchResult,
+  ProductInterestEntryRelation,
+  PaginationModel,
+  TranslationService,
+} from '@spartacus/core';
+import { tap, map } from 'rxjs/operators';
+import { ICON_TYPE } from '../../misc/icon/icon.model';
+@Component({
+  selector: 'cx-my-interests',
+  templateUrl: './my-interests.component.html',
+})
+export class MyInterestsComponent implements OnInit, OnDestroy {
+  iconTypes = ICON_TYPE;
+  private DEFAULT_PAGE_SIZE = 10;
+  private sortMapping = {
+    byNameAsc: 'name:asc',
+    byNameDesc: 'name:desc',
+  };
+
+  sort = 'byNameAsc';
+  sortOptions = [
+    {
+      code: 'byNameAsc',
+      selected: false,
+    },
+    {
+      code: 'byNameDesc',
+      selected: false,
+    },
+  ];
+  pagination: PaginationModel;
+
+  interests$: Observable<ProductInterestSearchResult>;
+  getInterestsloading$: Observable<boolean>;
+  getRemoveInterestsloading$: Observable<boolean>;
+
+  constructor(
+    private productInterestService: UserInterestsService,
+    private translationService: TranslationService
+  ) {}
+
+  ngOnInit() {
+    this.productInterestService.clearProductInterests();
+    this.interests$ = this.productInterestService
+      .getAndLoadProductInterests(this.DEFAULT_PAGE_SIZE)
+      .pipe(
+        tap(
+          interests =>
+            (this.pagination = {
+              currentPage: interests.pagination.page,
+              pageSize: interests.pagination.count,
+              totalPages: interests.pagination.totalPages,
+              totalResults: interests.pagination.totalCount,
+              sort: 'byNameAsc',
+            })
+        )
+      );
+    this.getInterestsloading$ = this.productInterestService.getProdutInterestsLoading();
+    this.getRemoveInterestsloading$ = this.productInterestService.getRemoveProdutInterestLoading();
+  }
+
+  removeInterest(result: ProductInterestEntryRelation): void {
+    this.productInterestService.removeProdutInterest(result);
+  }
+
+  getSortLabels(): Observable<{ byNameAsc: string; byNameDesc: string }> {
```

Comment:
This should be a property, if it's used in a template. Function will be recalculaten in each change detection cycle, and there will be new observable created each time.

Reply:
Fixed.